ConfigurationSystem/Agent/RucioSynchronizer.py

```diff
@@ -0,0 +1,359 @@
+                        for key in dictRSEAttributes:
+                            self.log.info('On %s, setting %s : %s' % (se, key, dictRSEAttributes[key]))
+                            client.add_rse_attribute(se, key, value=dictRSEAttributes[key])
+                        client.set_local_account_limit('root', se, 100000000000000000)
+                        if se.find('TMP-SE') > -1:
+                            client.add_rse_attribute(se, 'TMP-SE', True)
+                            client.set_local_account_limit(account, se, 100000000000000000)
+
+                    # Create the protocols
+                    try:
+                        protocols = client.get_protocols(se)
+                    except RSEProtocolNotSupported as err:
+                        self.log.info('Cannot get protocols for %s : %s' % (se, err))
+                        protocols = []
+                    existing_protocols = []
+                    for prot in protocols:
+                        existing_protocols.append((str(prot['scheme']), str(prot['hostname']), str(prot['port']), str(prot['prefix'])))
+                    protocols_to_create = []
+                    for cnt in range(len(result['Value'][se])):
+                        params = result['Value'][se][cnt]
+                        prot = (str(params['scheme']), str(params['hostname']), str(params['port']), str(params['prefix']))
+                        protocols_to_create.append(prot)
+                        if prot not in existing_protocols and prot[0] in valid_protocols:
+                            # The protocol defined in Dirac does not exist in Rucio. Will be created
+                            self.log.info('Will create new protocol %s://%s:%s%s on %s' % (params['scheme'], params['hostname'], params['port'], params['prefix'], se))
+                            try:
+                                client.add_protocol(rse=se, params=params)
+                            except Duplicate as err:
+                                self.log.info('Protocol %s already exists on %s' % (params['scheme'], se))
+                            except Exception as err:
+                                self.log.error('Cannot create protocol on RSE %s : %s' % (se, str(err)))
+                        else:
+                            update = False
+                            for protocol in protocols:
+                                if prot == (str(protocol['scheme']), str(protocol['hostname']), str(protocol['port']), str(protocol['prefix'])):
+                                    # Check if the protocol defined in Dirac has the same priority as the one defined in Rucio
+                                    for domain in ['lan', 'wan']:
+                                        for activity in ['read', 'write', 'delete']:
+                                            if params['domains'][domain][activity] != protocol['domains'][domain][activity]:
```

Comment:
Can this just be `if params['domains'] != protocol['domains']:` or are there other keys in `domains` that need to be ignored?